apps/src/aichat/aichatCompletionApi.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Role,
+  ChatCompletionMessage,
+  AiCustomizations,
+  ChatContext,
+  AiCustomizationsForBackend,
+} from './types';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import Lab2Registry from '@cdo/apps/lab2/Lab2Registry';
+
+const CHAT_COMPLETION_URL = '/aichat/chat_completion';
+
+/**
+ * This function sends a POST request to the aichat completion backend controller.
+ */
+export async function postAichatCompletionMessage(
+  newMessage: string,
+  storedMessages: AichatCompletionMessage[],
+  aiCustomizations: AiCustomizationsForBackend,
+  chatContext: ChatContext
+): Promise<AichatCompletionMessage | null> {
+  const payload = {
+    newMessage,
+    storedMessages,
+    aiCustomizations,
+    chatContext,
+  };
+  const response = await HttpClient.post(
+    CHAT_COMPLETION_URL,
+    JSON.stringify(payload),
+    true,
+    {
+      'Content-Type': 'application/json; charset=UTF-8',
+    }
+  );
+  // For now, response will be null if there was an error.
+  if (response.ok) {
+    return await response.json();
+  } else {
+    return null;
+  }
+}
+
+/**
+ * This function formats chat completion messages and aiCustomizations, passes data
+ * to `postAichatCompletion`, then returns the status of the response and assistant message
+ * if successful.
+ * TODO: Awaiting details on how to format input for endpoint.
+ */
+export async function getAichatCompletionMessage(
```

Comment:
Nit: the `get` and `post` naming imply to me two different HTTP request types...could the `post` one be `requestAiChatCompletionMessage` (or something else you come up with 😆 )

Reply:
+1 - could we even just merge this into `postAichatCompletionMessage` since it's mainly just passing on parameters?

Reply:
I was wondering about combining this since we're formatting data on the backend. Will do - thanks to you both!